api/src/main/java/com/epam/pipeline/manager/cluster/writer/MonitoringStatsWriter.java

```diff
@@ -0,0 +1,150 @@
+    private static final String DISK_USAGE_HEADER_TEMPLATE = "%s_usage[%%]";
+    private static final String NETWORK_USAGE_IN_HEADER_TEMPLATE = "%s_in[bytes]";
+    private static final String NETWORK_USAGE_OUT_HEADER_TEMPLATE = "%s_out[bytes]";
+    private static final double HUNDRED_PERCENTS = 100.0;
+
+    public String convertStatsToCsvString(final List<MonitoringStats> stats) {
+        final StringWriter stringWriter = new StringWriter();
+        final CSVWriter csvWriter = new CSVWriter(stringWriter);
+        final List<String[]> allLines = extractTable(stats);
+        csvWriter.writeAll(allLines);
+        return stringWriter.toString();
+    }
+
+    private List<String[]> extractTable(final List<MonitoringStats> stats) {
+        final List<String[]> allLines = new ArrayList<>();
+        final MonitoringStatsHeader header = extractHeader(stats);
+        if (!header.getColumnNames().isEmpty()) {
+            allLines.add(header.getColumnNames().toArray(new String[0]));
+            final List<String[]> entities = stats.stream()
+                .map(stat -> createNewLine(header, stat))
+                .collect(Collectors.toList());
+            allLines.addAll(entities);
+            return allLines;
+        } else {
+            return Collections.emptyList();
+        }
+    }
+
+    private MonitoringStatsHeader extractHeader(final List<MonitoringStats> stats) {
+        if (stats.size() == 0) {
+            return MonitoringStatsHeader.EMPTY_HEADER;
+        }
+        final List<String> headerColumns = new ArrayList<>(COMMON_STATS_HEADER);
+        final List<String> disks = stats.stream()
+            .map(MonitoringStats::getDisksUsage)
+            .map(MonitoringStats.DisksUsage::getStatsByDevices)
+            .map(Map::keySet)
+            .flatMap(Set::stream)
+            .distinct()
+            .sorted()
+            .peek(disk -> {
+                headerColumns.add(String.format(DISK_TOTAL_HEADER_TEMPLATE, disk));
+                headerColumns.add(String.format(DISK_USAGE_HEADER_TEMPLATE, disk));
+            })
+            .collect(Collectors.toList());
+        final List<String> networkInterfaces = stats.stream()
+            .map(MonitoringStats::getNetworkUsage)
+            .map(MonitoringStats.NetworkUsage::getStatsByInterface)
+            .map(Map::keySet)
+            .flatMap(Set::stream)
+            .distinct()
+            .sorted()
+            .peek(netInterface -> {
+                headerColumns.add(String.format(NETWORK_USAGE_IN_HEADER_TEMPLATE, netInterface));
+                headerColumns.add(String.format(NETWORK_USAGE_OUT_HEADER_TEMPLATE, netInterface));
+            })
+            .collect(Collectors.toList());
+        return new MonitoringStatsHeader(disks, networkInterfaces, headerColumns);
+    }
+
+    private String[] createNewLine(final MonitoringStatsHeader header, final MonitoringStats stat) {
+        final String[] newLine = new String[header.getColumnNames().size()];
```

Comment:
In order to remove hardcoded indices, I'd suggest to use `List` to build CSV line and then convert it to array

Reply:
Done